test/services/sync/strategies/LDAPSyncerConsumer.test.js

```diff
@@ -0,0 +1,218 @@
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+
+const sinon = require('sinon');
+const { ObjectId } = require('mongoose').Types;
+
+const { LDAP_SYNC_ACTIONS } = require('../../../../src/services/sync/strategies/LDAPSyncer');
+const { LDAPSyncerConsumer } = require('../../../../src/services/sync/strategies/LDAPSyncerConsumer');
+const { BadRequest } = require('../../../../src/errors');
+const { SchoolRepo, ClassRepo, UserRepo } = require('../../../../src/services/sync/repo');
+
+const appPromise = require('../../../../src/app');
+
+const { expect } = chai;
+chai.use(chaiAsPromised);
+
+describe('Ldap Syncer Consumer', () => {
+	let app;
+	let sandbox;
+
+	before(async () => {
+		app = await appPromise;
+	});
+
+	beforeEach(() => {
+		sandbox = sinon.createSandbox();
+	});
+
+	afterEach(() => {
+		sandbox.restore();
+	});
+
+	describe('execute message: ', () => {
```

Comment:
missing test for unexpected message type

Reply:
We have test for that:

`		it('should throw a BadRequest for unknown action', async () => {
			const content = { action: 'UNKNOWN', syncId: 'someId', data: {} };
			const message = { content: JSON.stringify(content) };
			expect(ldapConsumer.executeMessage(message)).to.eventually.throw(BadRequest);
		});`